cmd/flagsaver/flagsaver_test.go

```diff
@@ -0,0 +1,67 @@
+package flagsaver
+
+import (
+	"testing"
+	"time"
+
+	"flag"
+)
+
+var (
+	intFlag      = flag.Int("int_flag", 123, "test integer flag")
+	strFlag      = flag.String("str_flag", "foo", "test string flag")
+	durationFlag = flag.Duration("duration_flag", 5*time.Second, "test duration flag")
+)
+
+// TestRestore checks that flags are saved and restore correctly.
+// Checks are performed on flags with both their default values and with explicit values set.
+// Only a subset of the possible flag types are currently tested.
+func TestRestore(t *testing.T) {
+	tests := []struct {
+		// Test name
```

Comment:
Nits addressed, but left the comments, since I'm not sure `oldValue` and `newValue` are completely self-explanatory (perhaps I just need better names?).